src/browser/rollbar.js

```diff
@@ -0,0 +1,269 @@
+  this.options = _.extend(true, defaultOptions, options);
+  var context = 'browser';
+  this.client = client || new Client(context, this.options);
+  this.init(this.client);
+}
+
+Rollbar.prototype.global = function(options) {
+  this.client.global(options);
+};
+
+Rollbar.prototype.configure = function(options) {
+  this.options = _.extend(true, {}, this.options, options);
+  this.client.configure(options);
+};
+
+Rollbar.prototype.init = function(client) {
+  addTransformsToNotifier(client.notifier);
+  addPredicatesToQueue(client.queue);
+};
+
+Rollbar.prototype.log = function() {
+  var item = this._createItem(arguments);
+  var uuid = item.uuid;
+  this.client.log(item);
+  return {uuid: uuid};
+};
+
+Rollbar.prototype.debug = function() {
+  var item = this._createItem(arguments);
+  var uuid = item.uuid;
+  this.client.debug(item);
+  return {uuid: uuid};
+};
+
+Rollbar.prototype.info = function() {
+  var item = this._createItem(arguments);
+  var uuid = item.uuid;
+  this.client.info(item);
+  return {uuid: uuid};
+};
+
+Rollbar.prototype.warn = function() {
+  var item = this._createItem(arguments);
+  var uuid = item.uuid;
+  this.client.warn(item);
+  return {uuid: uuid};
+};
+
+Rollbar.prototype.warning = function() {
+  var item = this._createItem(arguments);
+  var uuid = item.uuid;
+  this.client.warning(item);
+  return {uuid: uuid};
+};
+
+Rollbar.prototype.error = function() {
+  var item = this._createItem(arguments);
+  var uuid = item.uuid;
+  this.client.error(item);
+  return {uuid: uuid};
+};
+
+Rollbar.prototype.critical = function() {
+  var item = this._createItem(arguments);
+  var uuid = item.uuid;
+  this.client.critical(item);
+  return {uuid: uuid};
+};
+
+Rollbar.prototype.handleUncaughtException = function(message, url, lineno, colno, error, context) {
+  var item;
+  if (error && _.isType(error, 'error')) {
+    item = this._createItem([message, error, context]);
+  } else if (url && _.isType(url, 'error')) {
+    item = this._createItem([message, url, context]);
+  } else {
+    item = this._createItem([message, context]);
+    item.stackInfo = _.makeUnhandledStackInfo(
+      message,
+      url,
+      lineno,
+      colno,
+      error,
+      'onerror',
+      'uncaught exception',
+      errorParser
+    );
+  }
+  item.level = this.options.uncaughtErrorLevel;
+  this.client.log(item);
+};
+
+Rollbar.prototype.handleUnhandledRejection = function(reason, promise) {
+  var message = 'unhandled rejection was null or undefined!';
+  message = reason ? (reason.message || String(reason)) : message;
+  var context = (reason && reason._rollbarContext) || promise._rollbarContext;
+
+  var item;
+  if (reason && _.isType(reason, 'error')) {
+    item = this._createItem([message, reason, context]);
+  } else {
+    item = this._createItem([message, context]);
+    item.stackInfo = _.makeUnhandledStackInfo(
+      message,
+      '',
+      0,
+      0,
+      null,
+      'unhandledrejection',
```

Comment:
Those are actually different parameters, `unhandledrejection` here corresponds to `onerror` above. It is the mode value that gets put into the stack trace.